### PR TITLE
Zadanie3

### PR DESCRIPTION
**Pull request by Tomoo3**
Zadanie 3

### CODE REVIEW

**Review by Kostya-Iva — APPROVED** [review]
- Widzę podział zbiorów tylko na train oraz test, ale bez walidacyjnego
- Brakuje wizualizacji wyników uczenia
- Zaskakujące wyniki ilości czasu uczenia

**Plusy:**
- Dobra analiza wyników
- Rozsądne wnioski

**Comment by farqlia** [comment]
Zadanie jest poprawnie zrobione, poniżej moje uwagi

- funkcja "run_epoch" niepotrzebnie jest od nowa pisana dla każdego modelu, można było dodać jako parametry model, optimizer i criterion. To samo z SimpleCNN, można przekazać wartość dropout jako argument zamiast powtarzać definicję klasy
- brakuje niejako EDY do tych zdjęć (np. jak uzasadnić skalowanie wartością 255?), gdzie gdybyśmy mieli różną częstość klas, to metryka accuracy mogłaby nie być dobrym wyborem (to tak na przyszłość)
- można było zacząć od sieci konwolucyjnej z jedną conv, bo zwykle szukamy najmniejszego modelu, który dobrze działa (a tu widać, że na 1szej epoce mamy >0.9 acc); tym bardziej, że SimpleNN ma jedną warstwę (nie wliczając ostatniej)
- brakuje zbioru validacyjnego, który służyłby do badań, a testowy byłby użyty na końcu do najlepszego modelu (typowy błąd na poprzedniej lekcji)
- nie ma wizualizacji do przebiegu trenowania, co też łączy się z tym, że dla 5ciu epok nie widać efektu przetrenowania; warto by było dodać krzywe trenowania, żeby też porównać modele między sobą w jasny sposób, no i np. na 20 epok, żeby wiedzieć, co tam dalej się dzieje (jako element badania)

**Review by fidok15 — APPROVED** [review]
- fajnie że jest przejście z tensorflowa na pytorcha z zbiorem 
- nwm czemu najpeirw zrobiłeś CNN a potem dopiero SimpleCNN, ale ok
- brakuje mi sprawdzenia zachowania modelu, dla więceji niż 5 epok 
- brak pokazania, z którymi przypadkami model sobie nie radzi 
- trafne wnioski 
- w tym ostatnim modelu coś się niedobrzego stało że tak niskie acc wyszło
